### DISCUSSION TITLE
Do not update data if no data

### DISCUSSION THREAD

**Discussion by ErnestBrandi** [vercel/swr#668] [General] [answered]
Hi, I feel silly but I haven't found any clue on how to achieve such a task.
Should I use the `compare` function and return true is new data is empty ?
Should I rather leave this task to react and use `React.memo` to not update the view ?
Thanks

**Comment by shuding** [comment]
What do you mean by "if no data"? Does the API return `null`, `undefined` or `{}`?

(If the data returned doesn't change, a render won't be triggered so the view won't update)

**Reply by ErnestBrandi** [reply]
I mean any of those values... 
It's up to me what is the server response.
But, what is not up to me, is if this response will carry data or not (I might reach API quota limit and then get no data from it).

EDIT : what I meant was; the server could return this for instance `{ products: [] }` or this `{ data: null }`.

**Comment by ErnestBrandi** [comment, marked as answer]
So I ended up using the `compare` function like so : 
```
compare: (_old, _new) => {
        // first time we get data
        if (!_old || !_new) return false;
        // bad new version -> do not change stale data
        if (!_new.whatever || isEmpty(_new.whatever)) // isEmpty is a lodash function
          return true;
        return dequal(_old, _new);
      }
```
